WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,334 @@
+import Foundation
+import Yosemite
+import class WooFoundation.CurrencyFormatter
+
+
+protocol CardPresentPayments {
+
+    /// The adaptor/facade is intended to be created once when we switch stores, and to be a singleton.
+    /// We might as well make a new one when we switch stores though.
+    init(siteID: Int64)
+
+    /// `paymentScreenEventPublisher` provides a stream of events relating to a payment, including their view models,
+    /// for subscribers to display to the user. e.g. onboarding screens, connection progress, payment progress, card reader messages
+    /// We may want separate streams for these depending on the way we consume it
+    var paymentScreenEventPublisher: AnyPublisher<CardPresentPaymentEvent?, Never> { get }
+
+    /// Uses the existing code to connect to a reader, publishing intermediate events on the stream above as required.
+    /// The end result of the payment is returned via async/await
+    /// - Parameters:
+    ///   - order: The order to collect payment for
+    ///   - discoveryMethod: Allows specifying Tap to Pay or external reader. For POS, this would default to external.
+    /// - Returns: `CardPresentPaymentResult` for a success, failure, or cancellation.
+    func collectPayment(for order: Order,
+                        using discoveryMethod: CardReaderDiscoveryMethod) async -> CardPresentPaymentResult
+
+    func cancelPayment()
+
+    func createTestOrder() async throws -> Order
+
+}
+
+enum CardPresentPaymentResult {
+    case success(Order)
+    case failure(CardPaymentErrorProtocol)
+    case cancellation
+}
+
+enum CardPresentPaymentEvent {
+    case presentAlert(CardPresentPaymentsModalViewModel) // These are existing view models
+    case presentReaderList(_ readerIDs: [String])
+    case showOnboarding(_ onboardingViewModel: InPersonPaymentsViewModel) // These are existing view models
+}
+
+protocol Presenting: AnyObject {
+    func present(_ viewControllerToPresent: UIViewController, animated: Bool, completion: (() -> Void)?)
+    func present(_ viewControllerToPresent: UIViewController, animated: Bool)
+    func dismiss(animated: Bool, completion: (() -> Void)?)
+    func dismiss(animated: Bool)
+    func show(_ vc: UIViewController, sender: Any?)
+    var presentedViewController: UIViewController? { get }
+    var navigationController: UINavigationController? { get }
+}
+
+extension UIViewController: Presenting {
+    func present(_ viewControllerToPresent: UIViewController, animated: Bool) {
+        present(viewControllerToPresent, animated: animated, completion: nil)
+    }
+
+    func dismiss(animated: Bool) {
+        dismiss(animated: animated, completion: nil)
+    }
+}
+
+class NullPresenting: Presenting {
```

Comment:
Yep, it's so we can pass something in which we know won't do anything when it's told to present, because we're also handling it another way. This lets us keep the underlying implementation and interface unchanged.

It's in place of using a `UIViewController()` that we just create and don't put in the view heirarchy... which is wasteful and probably unpredictable. It results in annoying runtime warnings, too.